addons/website/static/tests/tours/snippet_images_wall.js

```diff
@@ -58,7 +58,7 @@ selectSignImageStep,
 }, {
     content: "Change the link of the image",
     trigger: ".snippet-option-ReplaceMedia [data-set-url] input",
-    run: "edit /contactus && blur",
+    run: "edit /contactus && click body",
```

Comment:
Also add a TODO here, before your change a blur or click body was not needed, see here before the HOOT pr:
https://github.com/odoo/odoo/blame/3a4883bde4f28b25d210fc339b0f98ad603fdb82/addons/website/static/tests/tours/snippet_images_wall.js#L61

Could you please check all `&& click` added on website files and add a `TODO` on the ones where you added it yourself, which was not there initially (before https://github.com/odoo/odoo/commit/29bd1ddbd4fd26afbbe709125200be20a0262967)?

The one where it was `text_blur`, it's ok to keep uncommented.

Reply:
Thanks @rdeodoo for checking all of this but I don't think we should go ahead with adding a bunch of TODO @pipu-odoo ... unless you think you will handle them quickly after the merge? :shrug: Or do we have any lead as to why adding a bunch of "click body" are necessary with the new system? Otherwise this may be a big hint that something is not perfect with the system in the first place no? :grimacing:

> Also worth noting, change and blur events were apparently not properly triggered in HOOT tour framework when a step is inside an input and that the step after is clicking on another element.
> 
> It should be the case now, as @detrouxdev explained that to @pipu-odoo which made some adaptation so the hoot framework reflects that correctly.

I am curious: how could it **not** be the case? Could you explain the changes that were made in this new system maybe? Apart from the syntax, why is the behavior different?

Thanks :pray:

Reply:
Hello, I think that a little more precision should actually be given for the edit() helper.
Unlike previously implemented in runningActionTourHelper's edit helper, hoot.edit() "only" clears the input and fills it (keydown, keyup). There are no change or input events that are triggered in hoot.edit(). To trigger a blur or change event, you must either press Enter, or click outside, or press Tab, etc.
In most of case, the next "click" action in a step tour is enough.
in a previous PR, I therefore replaced all text_blur with edit && click. In some places, edit alone was not enough. However, hoot.edit() has been improved and fixed in the meantime.

Reply:
indeed. https://github.com/odoo-dev/odoo/blob/4169c4d698328ccb494b53ec1a1acf33d88a79ba/addons/web/static/lib/hoot-dom/helpers/events.js#L1068